src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java

```diff
@@ -0,0 +1,64 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryCreateDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryUpdateDTO;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+@Validated
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/categories")
+public class FinancialTransactionCategoryController{
+
+    @GetMapping("/{id}")
+    ResponseEntity<FinancialTransactionCategoryDTO> getFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable Long id){
+        //TODO Necessary code implementation
+        FinancialTransactionCategoryDTO financialTransactionCategoryDTO = null;
+        return new ResponseEntity<>(financialTransactionCategoryDTO, HttpStatus.OK);
+    }
+
+    @GetMapping()
+    ResponseEntity<List<FinancialTransactionCategoryDTO>> getFinancialTransactionCategories(){
+        //TODO Necessary code implementation
+        List<FinancialTransactionCategoryDTO> financialTransactionCategoryDTOList = null;
+        return new ResponseEntity<>(financialTransactionCategoryDTOList, HttpStatus.OK);
+    }
+
+    @PostMapping()
+    public ResponseEntity<FinancialTransactionCategoryDTO> createFinancialTransactionCategory(@Valid @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO){
+        //TODO Necessary code implementation
+        FinancialTransactionCategoryDTO financialTransactionCategoryDTO = null;
+        return new ResponseEntity<>(financialTransactionCategoryDTO, HttpStatus.CREATED);
+    }
+
+    @PatchMapping("/{id}")
+    public ResponseEntity<FinancialTransactionCategoryDTO> updateFinancialTransactionCategory(@Min(1) @NotNull @PathVariable Long id,
+                                                                                              @Valid @RequestBody FinancialTransactionCategoryUpdateDTO financialTransactionCategoryUpdateDTO){
+        //TODO Necessary code implementation
+        FinancialTransactionCategoryDTO financialTransactionCategoryDTO = null;
+        return new ResponseEntity<>(financialTransactionCategoryDTO, HttpStatus.OK);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<FinancialTransactionCategoryDTO> deleteFinancialTransactionCategoryById(@Min(1) @NotNull @PathVariable Long id){
```

Comment:
Does return type should not be Void?

Reply:
Think on the controller should be:
**- `new ResponseEntity<>(HttpStatus.NO_CONTENT)`
- `new ResponseEntity<>(HttpStatus.OK)`**
- definitely not `ResponseEntity<FinancialTransactionCategoryDTO>`

In an analogous service method, the method should return `void` or `boolean`. 
-`void` because deleting method on the repository is not returning the result value. So it's the proper answer for the service method result.
- 'boolea=true' because if the JPA method does not return the operation result, it can be assumed to be successful. However, following this logic, you can also assume that it failed even though it didn't throw an exception. So type 'boolean' would be a wrong assumption though.

Reply:
done